src/Watchdog/Watchdog.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace AgDevelop\ForkingSupervisor\Watchdog;
+
+use DateTime;
+use DateTimeImmutable;
+use DateTimeInterface;
+use Exception;
+
+class Watchdog implements WatchdogInterface
+{
+    protected DateTimeInterface $lastOccupied;
+
+    public function __construct(
+        private ?int $maxUnoccupiedTime,
+        private ?int $maxAliveTime,
+        private ?DateTimeInterface $createdAt = null,
+    ) {
+        if (null === $this->createdAt) {
+            $this->createdAt = new DateTimeImmutable('now');
+        }
+
+        $this->lastOccupied = clone $this->createdAt;
+    }
+
+    /**
+     * @throws Exception
+     */
```

Comment:
W sensie gdzie to coś rzuca?

Reply:
Oczywiście technicznie nic nie wymusza na implementacji ale sugeruje konwencję.

Reply:
`new DateTimeImmutable()` rzuca

https://www.php.net/manual/en/datetimeimmutable.construct.php